kcp/controllers/watcher_controller.go

```diff
@@ -159,181 +173,58 @@ func (r *WatcherReconciler) HandleDeletingState(ctx context.Context, logger logr
-		return err
-	}
-	return nil
-}
-
-func (r *WatcherReconciler) updateSKRWatcherConfigForCR(ctx context.Context, obj *componentv1alpha1.Watcher) error {
-	return deploy.InstallWebhookOnAllSKRs(ctx, releaseName, obj, r.Client, r.SkrWatcherPath)
-}
-
-func (r *WatcherReconciler) deleteServiceMeshConfigForCR(ctx context.Context, obj *componentv1alpha1.Watcher) error {
-	namespace := obj.GetNamespace()
-	vsName := obj.GetName()
-	istioClientSet, err := istioclient.NewForConfig(r.RestConfig)
-	if err != nil {
-		return fmt.Errorf("failed to create istio client set from rest config(%s): %w", r.RestConfig.String(), err)
-	}
-	_, err = istioClientSet.NetworkingV1beta1().VirtualServices(namespace).Get(ctx, vsName, metav1.GetOptions{})
-	if err != nil && !errors.IsNotFound(err) {
-		return fmt.Errorf("failed to get istio virtual service: %w", err)
-	}
-	if errors.IsNotFound(err) {
-		// nothing to do
-		return nil
-	}
-	err = istioClientSet.NetworkingV1beta1().VirtualServices(namespace).Delete(ctx, vsName, metav1.DeleteOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to delete istio virtual service: %w", err)
-	}
-	return nil
-}
-
-func (r *WatcherReconciler) updateWatcherCRStatus(ctx context.Context, obj *componentv1alpha1.Watcher,
-	state componentv1alpha1.WatcherState, msg string,
+func (r *WatcherReconciler) updateWatcherCRStatus(ctx context.Context, obj *watcherv1alpha1.Watcher,
+	state watcherv1alpha1.WatcherState, msg string,
 ) error {
 	obj.Status.State = state
 	switch state { //nolint:exhaustive
-	case componentv1alpha1.WatcherStateReady:
-		util.AddReadyCondition(obj, componentv1alpha1.ConditionStatusTrue, msg)
+	case watcherv1alpha1.WatcherStateReady:
+		addReadyCondition(obj, watcherv1alpha1.ConditionStatusTrue, msg)
 	case "":
-		util.AddReadyCondition(obj, componentv1alpha1.ConditionStatusUnknown, msg)
+		addReadyCondition(obj, watcherv1alpha1.ConditionStatusUnknown, msg)
 	default:
-		util.AddReadyCondition(obj, componentv1alpha1.ConditionStatusFalse, msg)
+		addReadyCondition(obj, watcherv1alpha1.ConditionStatusFalse, msg)
 	}
 	return r.Status().Update(ctx, obj.SetObservedGeneration())
 }
 
-func (r *WatcherReconciler) updateWatcherCRErrStatus(ctx context.Context, logger logr.Logger, err error,
-	obj *componentv1alpha1.Watcher, errMsg string,
-) error {
-	logger.Error(err, errMsg)
-	apiErr := r.updateWatcherCRStatus(ctx, obj, componentv1alpha1.WatcherStateError, errMsg)
-	if apiErr != nil {
-		logger.Error(apiErr, "update request to API server failed")
-		return apiErr
-	}
-	return err
+func addReadyCondition(obj *watcherv1alpha1.Watcher, state watcherv1alpha1.WatcherConditionStatus, msg string) {
+	obj.Status.Conditions = append(obj.Status.Conditions, watcherv1alpha1.WatcherCondition{
+		Type:               watcherv1alpha1.ConditionTypeReady,
+		Status:             state,
+		Message:            msg,
+		LastTransitionTime: &metav1.Time{Time: time.Now()},
+	})
```

Comment:
this appends all the time, there should be only one Ready condition per resource